### PR TITLE
[NP-9063] Add MarkdownView

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

File: src/foam/u2/view/MarkdownView.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+foam.CLASS({
```
Comment (jlhughes) [inline src/foam/u2/view/MarkdownView.js:7]: header

File: src/foam/core/stdlib.js [hunk 2]
```diff
@@ -730,6 +730,37 @@ foam.LIB({
         if ( ! safe ) console.warn('performed foam.String.applyMask with unsafe inputs');
         return newString;
       }
+    },
+    {
```
Comment (kgrgreer) [inline src/foam/core/stdlib.js]: I don't think this is common enough to include in stdlib.

File: src/foam/u2/view/MarkdownView.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'MarkdownView',
+  extends: 'foam.u2.View',
+
+  classes: [
+      {
```
Comment (kgrgreer) [inline src/foam/u2/view/MarkdownView.js]: indentation is wrong. should be 2 spaces

**Comment by kgrgreer** [comment]
Very neat! (By which I mean "interesting", not actually "neat", as your indentation is off ;-) ).

File: src/foam/core/stdlib.js [hunk 4]
```diff
@@ -730,7 +730,7 @@ foam.LIB({
         if ( ! safe ) console.warn('performed foam.String.applyMask with unsafe inputs');
         return newString;
       }
-    }
+    },
```
Comment (kgrgreer) [inline src/foam/core/stdlib.js]: remove trailing comma

File: src/foam/u2/view/MarkdownView.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+          if (!m) continue;
+          ruleMatched = true;
+          pos += m[0].length;
+
+          console.log(rule.create.toString(), m);
+          yield rule.create(m);
+          break;
+        }
+
+        if (!ruleMatched) {
```
Comment (kgrgreer) [inline src/foam/u2/view/MarkdownView.js]: spacing is wrong

File: src/foam/core/stdlib.js (hunk 2 above)
Comment (KernelDeimos) [reply src/foam/core/stdlib.js]: moved it into MarkdownView

File: src/foam/u2/view/MarkdownView.js [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        const str = markup.slice(pos);
+
+        let ruleMatched = false;
+        for (rule of rules) {
+          const m = str.match(rule.pattern);
+          if (!m) continue;
+          ruleMatched = true;
+          pos += m[0].length;
+
+          console.log(rule.create.toString(), m);
```
Comment (kgrgreer) [inline src/foam/u2/view/MarkdownView.js]: Should this be here?

File: src/foam/u2/view/MarkdownView.js [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+        }
+
+        if (!ruleMatched) {
+          console.error('could not find any matches', str);
+          break;
+        }
+      }
+    },
+    function dedent_ (str) {
+      console.log('dedent on ', str)
```
Comment (kgrgreer) [inline src/foam/u2/view/MarkdownView.js]: remove

File: src/foam/u2/view/MarkdownView.js (hunk 5 above)
Comment (KernelDeimos) [reply src/foam/u2/view/MarkdownView.js]: oh shoot, I used the formatter in vscode when I fixed the spacing, I shouldn't have done that

**Pull request merged**